modules/homebrew.nix

```
@@ -1,219 +1,751 @@
+      };
+      no_binaries = mkNullOrBoolOption {
+        description = "Whether to disable linking of helper executables.";
+      };
+
+      brewfileLine = mkInternalOption { type = types.nullOr types.str; };
+    };
+
+    config =
+      let
+        sCfg = mkProcessedSubmodConfig config;
+      in
+      {
+        brewfileLine =
+          if sCfg == { } then null
+          else "cask_args ${mkBrewfileLineOptionsListString sCfg}";
+      };
+  };
+
+  brewOptions = { config, ... }: {
+    options = {
+      name = mkOption {
+        type = types.str;
+        description = "The name of the formula to install.";
+      };
+      args = mkOption {
+        type = with types; nullOr (listOf str);
+        default = null;
+        description = ''
+          Arguments flags to pass to <command>brew install</command>. Values should not include the
+          leading <literal>"--"</literal>.
+        '';
+      };
+      conflicts_with = mkOption {
+        type = with types; nullOr (listOf str);
+        default = null;
+        description = ''
+          List of formulae that should be unlinked and their services stopped (if they are
+          installed).
+        '';
+      };
+      restart_service = mkOption {
+        type = with types; nullOr (either bool (enum [ "changed" ]));
+        default = null;
+        description = ''
+          Whether to run <command>brew services restart</command> for the formula and register it to
+          launch at login (or boot). If set to <literal>"changed"</literal>, the service will only
+          be restarted on version changes.
+
+          Homebrew's default is <literal>false</literal>.
+        '';
+      };
+      start_service = mkNullOrBoolOption {
+        description = ''
+          Whether to run <command>brew services start</command> for the formula and register it to
+          launch at login (or boot).
+
+          Homebrew's default is <literal>false</literal>.
+        '';
+      };
+      link = mkNullOrBoolOption {
+        description = ''
+          Whether to link the formula to the Homebrew prefix. When this option is
+          <literal>null</literal>, Homebrew will use it's default behavior which is to link the
+          formula if it's currently unlinked and not keg-only, and to unlink the formula if it's
+          currently linked and keg-only.
+        '';
+      };
+
+      brewfileLine = mkInternalOption { type = types.nullOr types.str; };
+    };
+
+    config =
+      let
+        sCfg = mkProcessedSubmodConfig config;
+        sCfgSubset = removeAttrs sCfg [ "name" "restart_service" ];
+      in
+      {
+        brewfileLine =
+          "brew ${sCfg.name}"
+          + optionalString (sCfgSubset != { }) ", ${mkBrewfileLineOptionsListString sCfgSubset}"
+          # We need to handle the `restart_service` option seperately since it can be either bool
+          # or the string value ":changed" the Brewfile.
```

```suggestion
          # or the string value "changed".
```

Not sure this is correct but the sentence as-is seems broken.

---

Thanks for catching this, will fix.